photutils/psf/models.py

```diff
@@ -15,7 +17,271 @@
+        model.
+
+    kwargs : dict, optional
+
+        Additional optional keyword arguments to be passed directly to the
+        `compute_interpolator` method. Possible values are:
+
+        - **degree** : int, tuple, optional
+            Degree of the interpolating spline. A tuple can be used to provide
+            different degrees for the X- and Y-axes. Default value is degree=3.
+
+        - **s** : float, optional
+            Non-negative smoothing factor. Default value s=0 corresponds to
+            interpolation.
+            See :py:class:`~scipy.interpolate.RectBivariateSpline` for more
+            details.
+
+    """
+    flux = Parameter(description='Total flux of the image.', default=None)
+    x_0 = Parameter(description='Shift along the X-axis relative to the '
+                    'origin.', default=0.0)
+    y_0 = Parameter(description='Shift along the Y-axis relative to the '
+                    'origin.', default=0.0)
+
+    def __init__(self, data, flux=flux.default,
+                 x_0=x_0.default, y_0=y_0.default,
+                 origin=None, weights=None, fillval=0.0, kwargs={}):
+        """
+        """
+        self._fillval = fillval
+        self._weights = weights
+
+        # compute flux and normalize data so that sum(data) = 1:
+        tflux = np.sum(data, dtype=np.float64)
+        if tflux == 0.0 or not np.isfinite(tflux):
+            tflux = 1.0
+        if flux is None:
+            flux = tflux
+        self._ndata = data / tflux
+
+        # set input image related parameters:
+        self._ny, self._nx = data.shape
+
+        # find origin of the coordinate system in image's pixel grid:
+        self.origin = origin
+
+        super(Discrete2DModel, self).__init__(flux, x_0, y_0)
+
+        # define interpolating spline:
+        self.compute_interpolator(kwargs)
+
+    @property
+    def ndata(self):
+        """Normalized model data such that sum of all pixels is 1."""
+        return self._ndata
+
+    @property
+    def data(self):
+        """Model data such that sum of all pixels is equal to flux."""
+        return (self.flux.value * self._ndata)
+
+    @property
+    def weights(self):
+        """
+        Weights of image data. When setting weights, :py:class:`numpy.ndarray`
+        or `None` may be used.
+        """
+        return self._weights
+
+    @weights.setter
+    def weights(self, weights):
+        self._weights = weights
+
+    @property
+    def shape(self):
+        """A tuple of dimensions of the data array in numpy style (ny, nx)."""
+        return self._ndata.shape
+
+    @property
+    def nx(self):
+        """Number of columns in the data array."""
+        return self._nx
+
+    @property
+    def ny(self):
+        """Number of rows in the data array."""
+        return self._ny
+
+    @property
+    def origin(self):
+        """
+        A tuple of `x` and `y` coordinates of the origin of the coordinate
```

Comment:
I'm 100% sure about this. :smile:  Double backticks are needed, otherwise it'll be an unresolved link.  Double backticks will change the rendered font, which is probably what was intended to set it apart from normal prose.